controllers/tokenupload_controller.go

```diff
@@ -61,100 +62,101 @@ type TokenUploadReconciler struct {
 }
 
 func (r *TokenUploadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-
 	lg := log.FromContext(ctx)
-	uploadSecretList := &corev1.SecretList{}
+	lg.V(logs.DebugLevel).Info("starting reconciliation")
+	defer logs.TimeTrackWithLazyLogger(func() logr.Logger { return lg }, time.Now(), "Reconcile Upload Secret")
+
+	uploadSecret := &corev1.Secret{}
+
+	if err := r.Get(ctx, req.NamespacedName, uploadSecret); err != nil {
+		if errors.IsNotFound(err) {
+			lg.V(logs.DebugLevel).Info("upload secret already gone from the cluster. skipping reconciliation")
+		}
+		return ctrl.Result{}, fmt.Errorf("failed to load the upload secret from the cluster: %w", err)
+	}
+
+	if uploadSecret.DeletionTimestamp != nil {
```

Comment:
For this case it would be rather good to make it deleted ASAP, just immediately after reading ... unfortunately it wont happen :) (not sufficient)

Reply:
We cannot guarantee that the reconciler won't be triggered between the time the secret is marked for deletion and actually deleted. This check is always there just to bail out quickly in a situation where the secret is in the process of being deleted (but the process is waiting for some condition that is outside of our responsibility).